### PR TITLE
F/wep 155 action on comment mutation

### PR DESCRIPTION
**Pull request by bishoymelek**

### CODE REVIEW

File: packages/api/src/graphql/mutation.private.ts [hunk 1]
```diff
@@ -737,6 +740,32 @@ export const GraphQLAdminMutation = new GraphQLObjectType<undefined, Context>({
         authorise(CanPublishPage, roles)
         return dbAdapter.page.unpublishPage({id})
       }
+    },
+
+    takeActionOnComment: {
```
Comment (roosnic1) [inline packages/api/src/graphql/mutation.private.ts:745]: I read in an article that this is considered bad practice. Every mutation should have one job and i agree with that philosophy. I suggest you create two mutations called `approveComment` and `rejectComment`. The database adapter method on the other hand should be simple and can be called `updateComment`.

File: packages/api/src/graphql/mutation.private.ts [hunk 2]
```diff
@@ -737,6 +740,32 @@ export const GraphQLAdminMutation = new GraphQLObjectType<undefined, Context>({
...
+      async resolve(root, {id, input}, {authenticate, dbAdapter}) {
+        const {roles} = authenticate()
+        authorise(CanTakeActionOnComment, roles)
+
+        const {state, rejectionReason} = input
+        if (
+          [CommentState.PendingUserChanges, CommentState.Rejected].includes(state) &&
+          !rejectionReason
+        ) {
+          throw new Error('Rejection Reason is required')
```
Comment (roosnic1) [inline packages/api/src/graphql/mutation.private.ts:760]: If you throw errors in the GraphQL API they should extend `ApolloError`. Check the file `packages/api/src/error.ts`

**Review by roosnic1 — CHANGES_REQUESTED** [review]

File: packages/api/src/graphql/mutation.private.ts (hunk 2 above)
Comment (bishoymelek) [reply packages/api/src/graphql/mutation.private.ts]: After changing the mutation to multiple ones, we don't need this check anymore.

File: packages/api/src/graphql/mutation.private.ts (hunk 1 above)
Comment (bishoymelek) [reply packages/api/src/graphql/mutation.private.ts]: Agree, We currently have three mutations to approve, reject or request changes from the user.

**Review by roosnic1 — APPROVED** [review]
👍🏾

**Review by penk85 — APPROVED** [review]

**Pull request merged**